Userland/Libraries/LibWeb/HTML/TraversableNavigable.cpp

```diff
@@ -825,6 +831,139 @@ TraversableNavigable::HistoryStepResult TraversableNavigable::apply_the_history_
     return HistoryStepResult::Applied;
 }
 
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#checking-if-unloading-is-canceled
+TraversableNavigable::CheckIfUnloadingIsCanceledResult TraversableNavigable::check_if_unloading_is_canceled(
+    Vector<JS::Handle<Navigable>> navigables_that_need_before_unload,
+    JS::GCPtr<TraversableNavigable> traversable,
+    Optional<int> target_step,
+    Optional<UserNavigationInvolvement> user_involvement_for_navigate_events)
+{
+    // 1. Let documentsToFireBeforeunload be the active document of each item in navigablesThatNeedBeforeUnload.
+    Vector<JS::GCPtr<DOM::Document>> documents_to_fire_beforeunload;
```

Comment:
Should this be a `Vector<JS::Handle<DOM::Document>>`?
I don't think we can guarantee that documents stay alive through the below code since it may invoke arbitrary JS, right?

Reply:
Good point :^)